File: seed/csharp-model/trace/src/SeedTraceClient/V2/V3/Problem/GetGeneratedTestCaseFileRequest.cs
using System.Text.Json.Serialization;
using SeedTraceClient.V2.V3;

namespace SeedTraceClient.V2.V3;

public class GetGeneratedTestCaseFileRequest
{
    [JsonPropertyName("template")]
    public TestCaseTemplate? Template { get; init; }

    [JsonPropertyName("testCase")]
    public TestCaseV2 TestCase { get; init; }
}

File: seed/php-sdk/trace/src/V2/V3/Problem/Types/GetGeneratedTestCaseFileRequest.php
<?php

namespace Seed\V2\V3\Problem\Types;

use Seed\Core\Json\JsonSerializableType;
use Seed\Core\Json\JsonProperty;

class GetGeneratedTestCaseFileRequest extends JsonSerializableType
{
    /**
     * @var ?TestCaseTemplate $template
     */
    #[JsonProperty('template')]
    public ?TestCaseTemplate $template;

    /**
     * @var TestCaseV2 $testCase
     */
    #[JsonProperty('testCase')]
    public TestCaseV2 $testCase;

    /**
     * @param array{
     *   testCase: TestCaseV2,
     *   template?: ?TestCaseTemplate,
     * } $values
     */
    public function __construct(
        array $values,
    )
    {
        $this->template = $values['template'] ?? null;$this->testCase = $values['testCase'];
    }

    /**
     * @return string
     */
    public function __toString(): string {
        return $this->toJson();
    }
}
